src/session/session_api.c

```diff
@@ -346,6 +346,14 @@ __wt_session_close_internal(WT_SESSION_IMPL *session)
         WT_TRET(__wt_call_log_close_session(session));
 #endif
 
+    /* Free the old error message. */
+    __wt_free(session, session->err_info.err_msg_buf->mem);
+    __wt_free(session, session->err_info.err_msg_buf);
+
+    /* Make sure no new error messages are saved during the close call. */
+    WT_TRET(__wt_session_set_last_error(session, 0, WT_NONE, WT_ERROR_INFO_EMPTY));
```

Comment:
I wonder if we should just use memset to clear the whole struct or set it individually, instead of calling the function. What do you think